girder/api/rest.py

```diff
@@ -618,6 +647,9 @@ def endpointDecorator(self, *path, **params):
                 # Don't do any post-processing of static files
                 return val
 
+            if isinstance(val, types.GeneratorType):
+                val = list(val)
```

Comment:
This is new functionality.  Before (and still), if something wanted to stream results, it returned a function which would return a generator.  Now, it can alternately return a generator and it will be coerced into a list.

There were several places where we were manually coercing generators into lists (for instance, in find assetstores, getAssetstoreFiles, listing groups, etc).  This avoids having to wrap those in `list()` in individual endpoints.

If we want to further add a streaming style to output a generator to a json list, that would be new functionality (which should be another PR).